libkbfs/folder_block_ops.go

```diff
@@ -1299,8 +1299,8 @@ func (fbo *folderBlockOps) writeDataLocked(
 			de.EncodedSize = 0
 			// update the file info
 			de.Size += uint64(len(block.Contents) - oldLen)
-			fbo.deCache[file.tailPointer().ref()] = de
 		}
+		fbo.deCache[file.tailPointer().ref()] = de
```

Comment:
This PR plus the upcoming one for KBFS-1261 actually exposes the issue in any existing test, `test/TestSimpleTruncate`.  But I can make a more-unit-y test for it when I get a chance.